test/dap/step_back_test.rb

```diff
@@ -607,11 +607,19 @@ def test_step_back_works_correctly
             message: "Success",
             body: {
               variables: [
+                {
+                  name: "%self",
```

Comment:
@ko1 When stepping forward, `variables` responses contain both `%self` and `bar`. But currently when stepping backward, `%self` is missing for some reason.
This PR happened to make it include `%self` and `bar` when stepping backward as well. I think it's a reasonable behavior so I adopted the tests with it.